backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/RunVmCommand.java

```diff
@@ -969,6 +969,9 @@ private void setDedicatedCpus() {
         }
         List<VdsCpuUnit> vdsCpuUnits = getVdsManager().getCpuTopology().stream()
                 .filter(cpu -> cpu.getVmIds().contains(getVmId())).sorted().collect(Collectors.toList());
+        if (vdsCpuUnits.isEmpty()) {
+            return;
```

Comment:
what does it mean if we get here?

Reply:
this is just for safety. for example with the current pinning logic, we may return with empty list. this will result here with empty string (""). when recovering in engine restart we try to parse it and will get exception. this will result in endless loop trying to init the vdsManager instance(via vmsMonitoring).

Reply:
> for example with the current pinning logic, we may return with empty list. this will result here with empty string (""). 

that's what I was afraid of - so a vm cannot be set with empy topology, it has to have at least one CPU - if we end up with empty pinning for a vm with dedicated cpus for some reason we should rather fail to run the vm, not to proceed with empty pinning..

Reply:
But that should never happen. These hosts should be filtered out. It happens now because the pinning logic (online cpus PR). This is a safeguard. I do agree that VM with dedicated policy, having `vcpus number != pcpus to use` is wrong, as it will run a VM without the configuration it was set with. Maybe we should raise an error and stop the command. This shouldn't happen anyway.

Reply:
> IIUC, this change is not related to the fixing of resize and pin as described in the commit message - why not add the change to your other PR (#135)? You need to update that PR anyway as the change would be lost.

+1